### PR TITLE
Rework OAuth2 module

### PR DESCRIPTION
**Pull request by DamnClin**
Close https://github.com/jhipster/jhipster-lite/issues/1689

### CODE REVIEW

**Comment by DamnClin** [comment]
My current idea is to create an `Account` domain aggregate with: 

* Username
* Name
  * Firstname
  * Lastname
* Email
* Roles

With all those fields mandatory (with roles that can be empty) and provide tooling to read that from token and to expose an endpoint for the authenticated user.

What do you think?

**Comment by antarus** [comment]
I think that's enough at first. we can always upgrade later

**Comment by pascalgrimaud** [comment]
@DamnClin : yes it's exactly what should be done, a specific account context

**Comment by DamnClin** [comment]
Ok, I should be able to do that tomorrow

File: src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/security/oauth2/domain/OAuth2ModuleFactory.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        .add("WithUnauthenticatedMockUser.java");
+    //@formatter:on
+  }
+
+  private void appendDependencies(JHipsterModuleBuilder builder) {
+    builder
+      .javaDependencies()
+      .add(SRPING_GROUP, artifactId("spring-boot-starter-security"))
+      .add(SRPING_GROUP, artifactId("spring-boot-starter-oauth2-client"))
+      .add(SRPING_GROUP, artifactId("spring-boot-starter-oauth2-resource-server"))
```
Comment (DanielFran) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/mvc/security/oauth2/domain/OAuth2ModuleFactory.java:130]:
```suggestion
      .add(SPRING_GROUP, artifactId("spring-boot-starter-security"))
      .add(SPRING_GROUP, artifactId("spring-boot-starter-oauth2-client"))
      .add(SPRING_GROUP, artifactId("spring-boot-starter-oauth2-resource-server"))
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**